uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1025.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1025 {
+
+  /**
+   * Deletes some demographics fields where they are not in use
+   *
+   * @return bool
+   */
+  public function upgrade_1025() {
+
+    $this->deleteExtendDemographicFields([
+      'Ethnicity',
+      'Religion',
+      'Sexual_Orientation',
+    ]);
+
+    return TRUE;
+  }
+
+  /**
+   * Deletes Custom Demographic Fields only if they are not used
+   *
+   * @param $fieldsToDelete
+   */
+  private function deleteExtendDemographicFields($fieldsToDelete) {
+    $customGroup = civicrm_api3('CustomGroup', 'get', [
+      'name' => ['LIKE' => 'Extended_Demographics'],
+    ]);
+    $customGroup = array_shift($customGroup['values']);
+    $customFields = civicrm_api3('CustomField', 'get', [
+      'name' => ['IN' => $fieldsToDelete],
+    ]);
+    foreach ($customFields['values'] as $customField) {
+      $query = 'SELECT * FROM ' . $customGroup['table_name'] . ' WHERE ' . $customField['column_name'] . ' NOT LIKE "%Not Applicable%" AND ' . $customField['column_name'] . ' IS NOT NULL AND ' . $customField['column_name'] . ' <> ""';
+      $dao = CRM_Core_DAO::executeQuery($query);
+      $isCustomFieldUsed = $dao->fetchAll();
```

Comment:
I don't think this makes sense for this variable name, `fetchAll` is returning a list of entities. I think it would make sense if you put the `empty` check around this on this line instead, i.e.

```
$isCustomFieldUsed = !empty($dao->fetchAll());
```